modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/apps-search-results/ddm-template.ftl

```diff
@@ -34,122 +63,83 @@
 	}
 </style>
 
-<#if serviceLocator??>
-	<#assign assetCategoryLocalService = serviceLocator.findService("com.liferay.asset.kernel.service.AssetCategoryLocalService") />
-</#if>
-
 <#assign
-	searchContainer = cpSearchResultsDisplayContext.getSearchContainer()
-
-	COMMERCE_PRODUCT_CLASS_NAME = "com.liferay.commerce.product.model.CPDefinition"
-	MARKETPLACE_PRICE_VOCABULARY_ID = 449511429
+	productsList = restClient.get("/headless-commerce-admin-catalog/v1.0/products?pageSize=-1").items
+	filteredProducts = 0
 />
 
+<#list productsList as product>
+	<#list product.categories?filter(category -> stringUtil.equals(category.name, "App")) as category>
```

Comment:
Please, use a variable for this
```suggestion
    <#assign appCategories = product.categories?filter(category -> stringUtil.equals(category.name, "App")) 
    />
	<#list appCategories as category>
```, because you are using it more than one time in the code.

Reply:
Done